src/test/java/com/rtb/projectmanagementtool/user/UserDataTest.java

```diff
@@ -0,0 +1,114 @@
+package com.rtb.projectmanagementtool.user;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.dev.LocalDatastoreService;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.*;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class UserDataTest {
+  // User 1
+  private static final long userID1 = 1l;
+  private static final long AuthID1 = 1l;
+  private static final String name1 = "Anna";
+  private static final long year1 = 2023;
+  private static final ArrayList<String> majors1 = new ArrayList<>(Arrays.asList("Biology", "Gov"));
+  private static final long total1 = 3;
+
+  // User 2
+  private static final long userID2 = 2l;
+  private static final long AuthID2 = 2l;
+  private static final String name2 = "Tal";
+  private static final long year2 = 2023;
+  private static final ArrayList<String> majors2 =
+      new ArrayList<>(Arrays.asList("Comp Sci", "Earth Sciences"));
+  private static final long total2 = 0;
+
+  // User 3
+  private static final long userID3 = 3l;
+  private static final long AuthID3 = 3l;
+  private static final String name3 = "Eddie";
+  private static final long year3 = 2022;
+  private static final ArrayList<String> majors3 = new ArrayList<>(Arrays.asList("Film", "Econ"));
+  private static final long total3 = 10;
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setAutoIdAllocationPolicy(LocalDatastoreService.AutoIdAllocationPolicy.SEQUENTIAL));
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testCreateFromConstructor() {
+    // create user by passing paramets to constructor
+    UserData user1 = new UserData(userID1, AuthID1, name1, year1, majors1, total1);
+
+    // ensure values are stored correct
+    Assert.assertEquals("userID", userID1, user1.getUserID());
+    Assert.assertEquals("AuthID", AuthID1, user1.getAuthID());
+    Assert.assertEquals("userName", name1, user1.getUserName());
+    Assert.assertEquals("userYear", year1, user1.getUserYear());
+    Assert.assertEquals("userMajors", majors1, user1.getUserMajors());
+    Assert.assertEquals("userTotalCompTasks", total1, user1.getUserTotal());
+  }
+
+  @Test
+  public void testCreateFromEntity() {
```

Comment:
test failure: testCreateFromEntity(com.rtb.projectmanagementtool.user.UserDataTest): userID expected:<2> but was:<0>

Any ideas as to why this is the case?

Reply:
An entity's key isn't initialized until its added to datastore w/ datastore.put(entity). Before that's done entity.getKey().getId() will always return 0. You've set the entity's id variable in the test, but when the user object is created its id is set to the entity's key

Reply:
Finally caught that. Thank you!